internal/conf/config.go

```diff
@@ -24,6 +24,7 @@ var DefaultOperatorConfig = OperatorConfig{
 	SmbdPort:                  445,
 	MetricsExporterMode:       "disabled",
 	ImagePullPolicy:           "IfNotPresent",
+	DefaultNodeSelector:       "",
```

Comment:
We would probably need to add in the future validation layer to `OperatorConfig` member. It is better to detect illegal inputs (in this case, malformatted json) early on at boot time instead of upon first usage.

Reply:
Yeah, that's tricky because I also want the config layer to be fairly generic and not have intimate detailed knowledge of what the layers above want.

Here's a thought: what about calling into the planner function somewhere in the main.go file and checking the error there and then blocking startup at that layer?

Reply:
Sounds good (in a separate PR)

Reply:
Ohhh well... take a look, I just pushed it as an additional patch to this one.

Reply:
LGTM